rspotify-model/src/page.rs

```diff
@@ -1,11 +1,20 @@
 //! All kinds of page object
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
 
-use serde::{Deserialize, Serialize};
+fn vec_without_nulls<'de, D, T>(deserializer: D) -> Result<Vec<T>, D::Error>
```

Comment:
Would you add a unit test for this function?

Reply:
And could you also add a comment for this function to demonstrate why do we need this custom deserialization function?

Reply:
I’m working on it, and I’m wondering how we handle the mismatch between the total number of results and the length of the `Vec`. `Null` elements are counted in the `total` field, but they are not present in the vector.

Reply:
We could accept this and add a doc to the `search` endpoint to demonstrate this anomaly behaviour. In my perspective, the Spotify server shouldn't return a `null` item, it doesn't make sense, and it requires all the clients and library to handle this weird design. What could the client do to handle `null` item except for filter them out.